src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -148,25 +167,33 @@ def run_ior_with_nvme_fault(self, oclass=None):
         }
         kwargs.update(ior_kwargs)
         thread.append(threading.Thread(target=thread_run_ior, kwargs=kwargs))
-
         # Launch the IOR thread
         self.log.info("Start IOR thread : %s", thread)
         thread[0].start()
 
         # 4.
-        self.log_step("dmg storage set nvme-faulty and check led state")
+        self.log_step(
+            "Marking the {} device as faulty and verifying it is 'EVICTED' and its "
+            "LED is 'ON'".format(test_dev))
         if not self.check_result(self.set_device_faulty(test_dev), "EVICTED", "ON"):
+            # Ensure the faulty device is reset in tearDown
+            self.register_cleanup(self.reset_fault_device, device=test_dev)
             self.fail("#Result of set_device_fault, device not in expected EVICTED, ON state")
+        # Wait for rebuild to start
+        self.pool.wait_for_rebuild_to_start()
+        # Wait for rebuild to complete
+        self.pool.wait_for_rebuild_to_end()
         # check device state after set nvme-faulty
-        time.sleep(1)
         if not self.verify_dev_led_state(test_dev, "NORMAL", "ON"):
+            self.register_cleanup(self.reset_fault_device, device=test_dev)
             self.fail("#After set_device_fault, device not back to NORMAL, ON state")
 
         # 5.
-        self.log_step("IOR continue")
+        self.log_step("Waiting for IOR to complete")
+
         # 6.
-        self.log_step("IOR completed and check for error")
         thread[0].join()
+        self.log_step("IOR completed and check for error")
```

Comment:
```suggestion
        self.log_step("Waiting for IOR to complete")
```
Maybe we don't need step 5 (line 192) as it's just a log statement? Then we can move this line (196) to 212.

Reply:
Test step 5 is just for logging match with test plan.